src/cdk-lib/bedrock/agent.ts

```diff
@@ -0,0 +1,643 @@
+          resources: [this.role.roleArn],
+        }),
+        new iam.PolicyStatement({
+          actions: ['bedrock:CreateAgent'],
+          resources: ['*'],
+        }),
+        new iam.PolicyStatement({
+          actions: [
+            'bedrock:DeleteAgent',
+            'bedrock:UpdateAgent',
+            'bedrock:TagResource',
+          ],
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              service: 'bedrock',
+              resource: 'agent',
+              resourceName: '*',
+              arnFormat: cdk.ArnFormat.SLASH_RESOURCE_NAME,
+            }),
+          ],
+        }),
+      ],
+    });
+
+    this.agent.node.addDependency(agentCRPolicy);
+    this.agent.node.addDependency(crProvider);
+
+    NagSuppressions.addResourceSuppressions(
+      agentCRPolicy,
+      [
+        {
+          id: 'AwsSolutions-IAM5',
+          reason: "Bedrock CreateAgent can't be restricted by resource.",
+        },
+      ],
+      true,
+    );
+
+    const kbAssocCRPolicy = new iam.Policy(this, 'KBAssocCRPolicy', {
+      roles: [crProvider.role],
+      statements: [
+        new iam.PolicyStatement({
+          actions: [
+            'bedrock:AssociateAgentKnowledgeBase',
+            'bedrock:UpdateAgentKnowledgeBase',
+            'bedrock:DisassociateAgentKnowledgeBase',
+          ],
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              service: 'bedrock',
+              resource: 'agent',
+              resourceName: '*',
+              arnFormat: cdk.ArnFormat.SLASH_RESOURCE_NAME,
+            }),
+            cdk.Stack.of(this).formatArn({
+              service: 'bedrock',
+              resource: 'knowledge-base',
+              resourceName: '*',
+              arnFormat: cdk.ArnFormat.SLASH_RESOURCE_NAME,
+            }),
+          ],
+        }),
+      ],
+    });
+
+    NagSuppressions.addResourceSuppressions(
+      kbAssocCRPolicy,
+      [
+        {
+          id: 'AwsSolutions-IAM5',
+          reason: 'Bedrock Agent/KB associations have wildcards restricted to agents and kbs in the account.',
+        },
+      ],
+      true,
+    );
+
+    const kbAssociations: string[] = [];
+    for (let kb of props.knowledgeBases ?? []) {
+      const kbAssoc = new cdk.CustomResource(
+        this,
+        `KBAssoc-${kb.name}`,
+        {
+          serviceToken: crProvider.serviceToken,
+          resourceType: 'Custom::Bedrock-AgentKnowledgeBase',
+          properties: {
+            agentId: this.agentId,
+            knowledgeBaseId: kb.kbId,
+            description: kb.description,
+          },
+        },
+      );
+      kbAssoc.node.addDependency(kbAssocCRPolicy);
+      kbAssoc.node.addDependency(crProvider);
+      kbAssociations.push(kbAssoc.getAttString('changeId'));
+    }
+
+    const changeIds: string[] = [this.agent.getAttString('changeId')];
+
+    changeIds.push(...kbAssociations);
+
+    this.alias = new cdk.CustomResource(
```

Comment:
Why do you forcibly create an Alias? This is not needed and the agent can be used even without alias. This should be optional and run only if Alias is specified

Reply:
Ok. As discussed, I will move this into an `addAlias` method.